### PR TITLE
Update configuration to use Archaius

### PR DESCRIPTION
**Pull request by akbarahmed**
I am still pushing more updates into this PR. I will ping you when the PR is ready to be merged.

These updates are converting the remaining configuration to use Archaius.

### CODE REVIEW

File: dynomitemanager/src/main/java/com/netflix/dynomitemanager/defaultimpl/DynomiteManagerConfiguration.java [hunk 1]
```diff
@@ -87,24 +101,22 @@ public int getIntProperty(String key, int defaultValue) {
 
     public static final String LOCAL_ADDRESS = "127.0.0.1";
 
-    private static final String CONFIG_DYN_HOME_DIR = DYNOMITEMANAGER_PRE + ".dyno.home";
-    private static final String CONFIG_DYN_START_SCRIPT = DYNOMITEMANAGER_PRE + ".dyno.startscript";
-    private static final String CONFIG_DYN_STOP_SCRIPT = DYNOMITEMANAGER_PRE + ".dyno.stopscript";
+    private static final String CONFIG_DYNOMITE_INSTALL_DIR = DYNOMITE_PROPS + ".install.dir";
```
Comment (ipapapa) [inline dynomitemanager/src/main/java/com/netflix/dynomitemanager/defaultimpl/DynomiteManagerConfiguration.java]: what is `DYNOMITE_PROPS`? We usually add the same string `DYNOMITEMANAGER_PRE` to all FPs.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Dynomite-manager
+# Dynomite Manager
```
Comment (ipapapa) [inline README.md]: I am curious about the name change... Why `Dynomite Manager` is better than `Dynomite-Manager`. The latter seems to provide a single name to the project. With your proposal, it is hard to understand if the project is called `Manager` or `Dynomite Manager`. I am not against it, but was wondering the reasoning ;)

Comment (akbarahmed) [inline README.md]: IMHO, a space reads better.

The repo is obviously `dynomite-manager`. 

However, I thought the product was `Dynomite Manager` (i.e. a "Manager" for Dynomite). For example, its DSE OpsCenter (DataStax) or Cloudera Manager. Most companies use a space with capitalization for each word.

I think the following capitalization and spacing makes for a good product family:

- Dynomite
- Dynomite Manager
- Dyno

Comment (ipapapa) [inline README.md]: Makes sense. Thanks for the info.

**Comment by akbarahmed** [comment]
This PR is ready for review / merge. 

There are approx. 48 more properties that need to be updated to use Archaius. I'll rebase and continue after this PR is merged.

File: dynomitemanager/src/main/resources/dynomitemanager.properties [hunk 3]
```diff
@@ -92,8 +92,14 @@ dm.dynomite.storage.preconnect=true
 # ENV: DM_DYNOMITE_MULTI_DC
 dm.dynomite.multi.dc=true
 
-#florida.dyno.mbuf.size=16384
-#florida.dyno.allocated.messages=200000
+# Dynomite memory buffer (mbuf) size in bytes
+# OLD: florida.dyno.mbuf.size=16384
+# ENV: DM_DYNOMITE_MBUF_SIZE
+dm.dynomite.mbuf.size=16384
```
Comment (shailesh33) [inline dynomitemanager/src/main/resources/dynomitemanager.properties]: How will this change affect existing systems?

Comment (akbarahmed) [inline dynomitemanager/src/main/resources/dynomitemanager.properties]: I have kept backward compatibility with the environment variables.

However, I have broken backward compatibility with the FP names, although this is only an issue when a non-default value is set.

If we need to maintain backward compatibility then I can update each configuration method to first check the old property, and if it's null, then use Archaius.

Let me know if this approach sounds okay. If we need to maintain backward compatibility with the FP keys, then I'll make this update to each method.

Comment (ipapapa) [inline dynomitemanager/src/main/resources/dynomitemanager.properties]: The approach you followed sounds OK to me.

**Comment by ipapapa** [comment]
LGTM

File: dynomitemanager/src/main/resources/dynomitemanager.properties (hunk 3 above)
Comment (shailesh33) [inline dynomitemanager/src/main/resources/dynomitemanager.properties]: yannis explained me the situation here. LGTM. Hopefully other folks are not using these properties. How would we know if we are not breaking anything for anyone here?

Comment (akbarahmed) [inline dynomitemanager/src/main/resources/dynomitemanager.properties]: @shailesh33 We won't know unless we get complaints on GH issues.

However, I've been doing a good job of documenting the old vs. new property names in the [dynomitemanager.properties](https://github.com/DynomiteDB/dynomite-manager/blob/archaius_20161117/dynomitemanager/src/main/resources/dynomitemanager.properties) file.

In the properties file I have 3 lines for each property:

- OLD: the old property
- ENV: the environment variable used to override the FP
- then the new property

**Comment by shailesh33** [comment]
Looks good to me.
Thanks for answering.

**Pull request merged**